frontend/src/components/Block/Block.tsx

```diff
@@ -93,6 +80,7 @@ const Block = () => {
             session: activeSession
         });
         if (round) {
+            setRound({ ...round.next_round }); // Make a deep copy of the round as the 'round' object will be mutated by the updateActions
```

Comment:
As `updateActions` performs a `.shift()`, it mutates the object, which is (unwantedly) also reflected in the store.